presto/client.nim

```diff
@@ -14,6 +14,22 @@ import segpath, common, macrocommon, agent
 export httpclient, httptable, httpcommon, options, agent, httputils
 export SocketFlags
 
+when defined(metrics):
+  import metrics
+
+  declareGauge presto_client_response_status_count,
```

Comment:
this should be a counter - ie anything that monotonically increases (like a counter) should use `declareCounter`. Ditto `connect_time` and the like.

Gauges are for one-off samples - for example, how many peers are _currently_ connected - a number that goes both up and down.

The metrics are sampled at some interval - if `connect_time` is a gauge and two connections happen between two samples, the first will be lost / overwritten. With a counter, they add up and you get a correct number irrespective of the sampling interval.